subprojects/docs/src/docsTest/java/org/gradle/docs/samples/IntegrationTestSamplesExecutor.java

```diff
@@ -71,6 +73,11 @@ protected int run(String executable, List<String> args, List<String> flags, Outp
             .withToolchainDetectionEnabled()
             .withArguments(filteredFlags)
             .withTasks(args);
+
+        if (flags.stream().anyMatch(f -> f.equals(NO_STACKTRACE_CHECK))) {
```

Comment:
I think this should go up where we already extract the `warningMode` and the system property shouldn't be propagated to the actual build.

Reply:
Actually I prefer doing a bit refactoring work to get rid of the loop.